### PR TITLE
C51-219: Activities calendar fixes

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview

This PR fixes the following issues with the activity calendar:

* It fixes the calendar date status when the browser font size changes
* It adds a small spacing between the calendar date and the activities list dropdown
* It removes the loading screen of the case details when an activity in the calendar is updated

### Before
screen shot 2018-10-04 at 11 28 12 pm
Notice how the 1st and the 5th status icons look like oval instead of circles

### After
screen shot 2018-10-04 at 11 27 04 pm

### Technicals Details

The issue happens because the status element changes sizes depending on the browser setting. To avoid this the element was changed to a material icon:

```html
<button type="button" class="btn btn-default btn-sm"
  <!-- ... -->>
  <!-- ... -->
  <i class="material-icons">lens</i>
</button>
```

And the CSS:

```scss
.civicase__activities-calendar__day-status.uib-day {
  .material-icons {
    color: $crm-gray-matte;
    display: block;
    font-size: 6px;
    left: 50%;
    position: absolute;
    transform: translateX(-50%) translateY(3px);
    width: 6px;
  }
}

.civicase__activities-calendar__day-status--completed.uib-day .material-icons {
  color: $brand-success;
}

.civicase__activities-calendar__day-status--overdue.uib-day .material-icons {
  color: $brand-danger;
}

.civicase__activities-calendar__day-status--scheduled.uib-day .material-icons {
  color: $brand-info;
}
```

### Before
record

### After
record

### Technical details

The issue happens because the case details now employe a `areDetailsLoaded` property (introduced in this PR https://github.com/compucorp/uk.co.compucorp.civicase/pull/53) that determines if a loading screen should be displayed or not. Displaying this loading screen breaks the calendar popover, and it's not necessary to block the whole UI for the case details when a single activity has been updated.

To fix this, a conditional parameter `useLoadingScreen` was introduced to the case details refresh function. This new property can be used to display the loading screen conditionally:

```js
$scope.refresh = function (apiCalls, options) {
  options = _.extend({}, { useLoadingScreen: true }, options);

  if (options.useLoadingScreen) $scope.areDetailsLoaded = false;
  if (!_.isArray(apiCalls)) apiCalls = [];

  apiCalls.push(['Case', 'getdetails', caseGetParams()]);
  crmApi(apiCalls, true).then(function (result) {
    $scope.pushCaseData(result[apiCalls.length - 1].values[0]);
    $scope.areDetailsLoaded = true;
  });
};
```

The activities calendar was modified so the refresh function blocks the loading ui from showing:

```js
$scope.refresh = function () {
  $scope.refreshCallback([], { useLoadingScreen: false });
};
```

### CODE REVIEW

**Comment by deb1990** [comment]
after

@reneolivo looks like the refresh functionality still has some problem. It does not update correctly. But when I close the popup and open it again, it updates.

**Comment by deb1990** [comment]
Rest 2 issues are working fine.

**Comment by reneolivo** [comment]
@deb1990 I can't replicate the issue you are showing here. The closes I can get is when slowing down the connection, but still, there's a `"saving..."` message whilst the activity is updating. Once it's done the calendar is updated as well:

case-calendar

File: ang/civicase/CaseDetails.js [hunk 1]
```diff
@@ -148,13 +148,23 @@
...
+     * Refreshes the Case Details data
+     *
+     * @param {Array}   apiCalls extra api calls to load on refresh.
+     * @param {Object}  options aditional options to consider when refreshing the data.
+     * @param {Boolean} options.useLoadingScreen displays the loading screen while refreshing the data.
+     */
+    $scope.refresh = function (apiCalls, options) {
+      options = _.extend({}, { useLoadingScreen: true }, options);
 
+      if (options.useLoadingScreen) $scope.areDetailsLoaded = false;
```
Comment (deb1990) [inline ang/civicase/CaseDetails.js]: Please use curly brackets for if statements.

File: ang/civicase/CaseDetails.js [hunk 2]
```diff
@@ -148,13 +148,23 @@
...
       if (!_.isArray(apiCalls)) apiCalls = [];
```
Comment (deb1990) [inline ang/civicase/CaseDetails.js]: Please use curly brackets for if statements.

**Pull request merged**